### PR TITLE
MR-398 Multiple SORs

### PR DESCRIPTION
**Pull request by Asura5**

### CODE REVIEW

File: src/components/Property/RaiseWorkOrder/BudgetCodeItemView.js [hunk 1]
```diff
@@ -74,7 +74,8 @@ const BudgetCodeItemView = ({
 }
 
 BudgetCodeItemView.propTypes = {
-  budgetCodeId: PropTypes.string.isRequired,
+  budgetCodeId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    .isRequired,
```
Comment (LBHCallumM) [inline src/components/Property/RaiseWorkOrder/BudgetCodeItemView.js:78]: Is there a chance that this could cause side effects? Surely its better to better to have either string or number.

File: src/components/Property/RaiseWorkOrder/RateScheduleItemView.js [hunk 2]
```diff
@@ -18,12 +18,24 @@ const RateScheduleItemView = ({
...
+    if (arrayOfRateScheduleItemComponentIndexes.length > 1) {
+      if (
+        formState.rateScheduleItems.length <
+        arrayOfRateScheduleItemComponentIndexes.length
+      ) {
+        arrayOfRateScheduleItemComponentIndexes.pop()
+      }
+    }
+  }, arrayOfRateScheduleItemComponentIndexes)
+
```
Comment (LBHCallumM) [inline src/components/Property/RaiseWorkOrder/RateScheduleItemView.js:38]: This could be much cleaner if you negated the if statements. For example:

```suggestion
  useEffect(() => {
    if (arrayOfRateScheduleItemComponentIndexes.length === 0) return;
    if (formState.rateScheduleItems.length === arrayOfRateScheduleItemComponentIndexes.length) return;
      
    arrayOfRateScheduleItemComponentIndexes.pop();
  }, arrayOfRateScheduleItemComponentIndexes)

```

**Review by LBHCallumM — APPROVED** [review]

File: src/components/Property/RaiseWorkOrder/BudgetCodeItemView.js (hunk 1 above)
Comment (ange47rm) [reply src/components/Property/RaiseWorkOrder/BudgetCodeItemView.js:78]: I set this because there were warnings about the prop type being passed in. 
What you say makes sense, however what was happening was:
- if budgetCodeId was set to PropTypes.string.isRequired - the console Warning would complaining about the component being passed a number (instead of the expected string.
- I switched the PropTypes.string.isRequired to PropTypes.number.isRequired and then the warning was complaining about the value going it being a string and not a number.
Basically it'd complain either way (even if you'd expect the passed in value to be always the same type)... That's why to remove the warning I set it that way.
I agree it kind of goes against what we're trying to do here, defining types as we're making it more vague.

Comment (LBHCallumM) [reply src/components/Property/RaiseWorkOrder/BudgetCodeItemView.js:78]: Is it complaining because in some places the prop is a string, and other places it is a number? Surely a better solution would be to convert all of the numbers to strings

File: src/components/Property/RaiseWorkOrder/RateScheduleItemView.js (hunk 2 above)
Comment (ange47rm) [reply src/components/Property/RaiseWorkOrder/RateScheduleItemView.js:38]: To my eyes, I think the way Dave wrote it is easier to read but that's subjective

Comment (Asura5) [reply src/components/Property/RaiseWorkOrder/RateScheduleItemView.js:38]: Also, the logic is different. In the top example if arrayOfRateScheduleItemComponentIndexes.length > formState.rateScheduleItems.length, the pop won't happen, whereas in the reworked example it will
